File: Implementierung/Backend/src/main/java/edu/kit/informatik/dto/userdata/interactions/InteractionDto.java
package edu.kit.informatik.dto.userdata.interactions;


import edu.kit.informatik.model.userdata.interactions.Interaction;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.sql.Timestamp;

/**
 * Data-Transfer-Object zur Klasse {@link Interaction}
 *
 * @author uekai
 * @author ugqbo
 * @version 1.0
 */
@EqualsAndHashCode
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class InteractionDto implements Comparable<InteractionDto> {

    private String id;
    private String userId;
    private String timeStamp;
    private String sessionId;
    private String toParticipantId;
    private String fromParticipantId;
    private String categoryId;
    private Timestamp createdAt;
    private Timestamp updatedAt;

    @Override
    public int compareTo(InteractionDto o) {
        Timestamp thisInteraction = this.createdAt;
        Timestamp oInteraction = o.createdAt;

        return  thisInteraction.compareTo(oInteraction);
    }
}

File: Implementierung/frontend/paschi/src/dto/userdata/interactions/InteractionDto.ts
export class InteractionDto {

  id: string;
  userId: string;
  createdAt: string;
  updatedAt: string;
  timeStamp: string;
  sessionId: string;
  toParticipantId: string;
  fromParticipantId: string;
  categoryId: string;

  constructor(id: string, userId: string, createdAt: string, updatedAt: string, timeStamp: string, sessionId: string,
              toParticipantId: string, fromParticipantId: string, categoryId: string) {
    this.id = id;
    this.userId = userId;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.timeStamp = timeStamp;
    this.sessionId = sessionId;
    this.toParticipantId = toParticipantId;
    this.fromParticipantId = fromParticipantId;
    this.categoryId = categoryId;
  }
}
